### PR TITLE
HTML: test stylesheet loading text/css requirement

### PR DESCRIPTION
**Pull request by annevk**

### CODE REVIEW

**Comment by annevk** [comment]
I wrote this for https://bugzilla.mozilla.org/show_bug.cgi?id=562377. It seems this could use some cleanup in HTML as well around the requirements.

**Comment by annevk** [comment]
(I would have thought we test this elsewhere too, but I couldn't find anything quickly.)

**Review by zcorpan — APPROVED** [review]

**Comment by annevk** [comment]
@zcorpan thanks, do you happen to know if we have quirks tests for this? If not, I'll add more tests Monday or so.

File: html/semantics/document-metadata/the-link-element/link-style-error-01.html [hunk 1]
```diff
@@ -44,5 +45,44 @@
   elt.rel = "stylesheet";
   elt.href = "../../../../../common/css-red.txt";
   document.getElementsByTagName("head")[0].appendChild(elt);
-})
+});
```
Comment (domfarolino) [inline html/semantics/document-metadata/the-link-element/link-style-error-01.html]: nit: Would you be up for adding semicolons above as well to match here, and maybe use t.step_func_done() in the above preexisting tests?

**Review by domfarolino — APPROVED** [review]

File: html/semantics/document-metadata/the-link-element/link-style-error-01.html (hunk 1 above)
Comment (annevk) [reply html/semantics/document-metadata/the-link-element/link-style-error-01.html]: I try not to make stylistic changes at the same time as that messes with blame too much in my opinion. And then doing it as a separate commit often seems like too much effort... I might not be making the right tradeoffs though.

**Comment by annevk** [comment]
https://github.com/whatwg/html/issues/3457 is probably the best HTML Standard issue to track improvements at.

**Comment by annevk** [comment]
I don't think we should block landing this on those happening though.

**Comment by gsnedders** [comment]
"limitedd" (in the commit message) isn't how you spell limited. :)

Also: 🎉 for having tests for this.

**Comment by annevk** [comment]
* https://bugs.chromium.org/p/chromium/issues/detail?id=888508
* https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/19085222/
* https://bugs.webkit.org/show_bug.cgi?id=189913

File: html/semantics/document-metadata/the-link-element/link-style-error-quirks.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE HTML PUBLIC "-//Sun Microsystems Corp.//DTD HotJava Strict HTML//" "">
```
Comment (zcorpan) [inline html/semantics/document-metadata/the-link-element/link-style-error-quirks.html]: Can you replace this with a comment saying "quirks mode"?

File: html/semantics/document-metadata/the-link-element/link-style-error-limited-quirks.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+<title>link: error events in limited quirks mode</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src=/common/get-host-info.sub.js></script>
+<div id="log"></div>
+<script>
+[
+ ["same-origin", "resources/css.py"],
+ ["cross-origin", get_host_info().HTTP_REMOTE_ORIGIN + "/html/semantics/document-metadata/the-link-element/resources/css.py"]
+].forEach(originType => {
```
Comment (zcorpan) [inline html/semantics/document-metadata/the-link-element/link-style-error-limited-quirks.html]: Please deduplicate the code by moving it into a separate JS file.

**Review by zcorpan — CHANGES_REQUESTED** [review]

File: html/semantics/document-metadata/the-link-element/link-style-error-quirks.html (hunk 2 above)
Comment (annevk) [reply html/semantics/document-metadata/the-link-element/link-style-error-quirks.html]: That's a lot less fun.

File: html/semantics/document-metadata/the-link-element/link-style-error-limited-quirks.html (hunk 3 above)
Comment (annevk) [reply html/semantics/document-metadata/the-link-element/link-style-error-limited-quirks.html]: I guess that's reasonable. Do you think it's fine to make it depend on document.compatMode or do you think it should be something you call externally?

The other thing I was thinking of also testing is `@import`, but how that affects load/error events isn't really defined yet.

Comment (zcorpan) [reply html/semantics/document-metadata/the-link-element/link-style-error-limited-quirks.html]: Using `document.compatMode` should be fine.

**Review by zcorpan — APPROVED** [review]

**Review by zcorpan — APPROVED** [review]

**Comment by annevk** [comment]
Thanks @zcorpan. I might wait a bit for @bzbarsky to take a look if he wants, since I'm not entirely sure about the `@import` stuff, but if that doesn't come to pass I need to at least file bugs.

**Comment by bzbarsky** [comment]
The test is loading urls like "resources/css.py...." from a sheet that's loaded from a data: URL.  Those relative URIs are resolved relative to the data: URL, which fails, so the load is not performed at all.  Whether this should trigger the error event is an interesting question, but at the moment the test is not testing what it thinks it's testing.

**Pull request merged**